busybox/busybox.py

```diff
@@ -26,21 +26,23 @@ def harness(dut: Dut, ctx: TestContext, result: TestResult):
         parsed = dut.match.groupdict()
 
         if idx == 2:
-            msg.append("\t\t" + parsed["line"])
-            continue
+            line_msg = parsed["line"] + " "
```

Comment:
I don't really know how the full output from the succesful/failed tescases looks like. Maybe we can put sample output as a mulitiline variable / comment after imports (to be used for pytest harness unit test in the near future hopefully).

From your code I suspect the output might be:
```
FAIL: test-name-1
why-failed-line-1
why-failed-line-2
msg-for-test-2
msg-for-test-2
PASS: test-name-2
```

If so - we can't assign message lines to the testcases in any sensible way (your code would append all 4 lines to `test-name-1`, but that's ok if the input is really like that) - but we should mention in comment that the messages might be invalid in that case.

Reply:
After checking, the output can appear in several ways. It might be produced during the test. In the event of a failure, lines can be printed both during the test when something fails and after the test. However, when printing strictly informational messages, they are preceded by `======================` to distinguish them.

e.g.:
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/116297248/46c6a07c-2933-43c2-af71-eae36689d6d9)
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/116297248/cb1ec17f-b48a-416c-b0bc-71c760986ab8)

Reply:
I've checked the busybox testsuite and it seems seems to be true only for newstyle tests (oldstyle tests don't print `===` nor messages before the test result - see `mkdir` for example). I don't see how to distinguish them in harness, maybe we should patch busybox testsuite framework instead?